okhttp/src/main/java/com/squareup/okhttp/CertificatePinner.java

```diff
@@ -123,17 +123,45 @@ private CertificatePinner(Builder builder) {
    * pinned for {@code hostname}. OkHttp calls this after a successful TLS
    * handshake, but before the connection is used.
    *
+   * @return {@code true} if hostname has pinned certificates and check passed,
```

Comment:
Yeah. I think if you want to make this more testable, extract a package-protected method that returns a `List<ByteString>` of matching pins? (The method does two things: it finds pins to match against, and then it matches against them. You want to test each independently.)